evadb/executor/vector_index_scan_executor.py

```diff
@@ -124,20 +123,24 @@ def _evadb_vector_index_scan(self, *args, **kwargs):
                 f"The index {self.index_name} returned only {num_required_results} results, which is fewer than the required {self.limit_count.value}."
             )
 
-        res_row_list = [None for _ in range(num_required_results)]
+        final_df = pd.DataFrame()
+        row_num_df = pd.DataFrame({"row_num_np": row_num_np})
         for batch in self.children[0].exec(**kwargs):
             column_list = batch.columns
             if not row_num_col_name:
                 row_num_alias = get_row_num_column_alias(column_list)
                 row_num_col_name = "{}.{}".format(row_num_alias, ROW_NUM_COLUMN)
 
-            # Nested join.
-            for _, row in batch.frames.iterrows():
-                for idx, row_num in enumerate(row_num_np):
-                    if row_num == row[row_num_col_name]:
-                        res_row = dict()
-                        for col_name in column_list:
-                            res_row[col_name] = row[col_name]
-                        res_row_list[idx] = res_row
+            result_df = pd.merge(
```

Comment:
Instead of doing O(n) of merging, will we get better performance if get all batches from the child and do merging only once?

Reply:
> Instead of doing O(n) of merging, will we get better performance if get all batches from the child and do merging only once?

Thanks for the suggestion, I have also made changes to not add child frames into the result df before merging to avoid unnecessary processing. The speedup is 2X now.